src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOf.Fixer.cs

```diff
@@ -0,0 +1,126 @@
+            SyntaxNode root = await doc.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is SyntaxNode expression)
+            {
+                SemanticModel semanticModel = await doc.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var compilation = semanticModel.Compilation;
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemString, out INamedTypeSymbol? stringType) ||
+                    !compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemChar, out INamedTypeSymbol? charType) ||
+                    !compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison, out INamedTypeSymbol? stringComparisonType))
+                {
+                    return;
+                }
+
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                // Not offering a code-fix for the variable declaration case
+                if (!(operation is IBinaryOperation binaryOperation))
+                {
+                    return;
+                }
+
+                IInvocationOperation invocationOperation;
+                IOperation otherOperation;
+                if (binaryOperation.LeftOperand is IInvocationOperation invocationOperationOperand)
+                {
+                    invocationOperation = invocationOperationOperand;
+                    otherOperation = binaryOperation.RightOperand;
+                }
+                else
+                {
+                    invocationOperation = (IInvocationOperation)binaryOperation.RightOperand;
+                    otherOperation = binaryOperation.LeftOperand;
+                }
+
+                var instanceOperation = invocationOperation.Instance;
+
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: MicrosoftNetCoreAnalyzersResources.PreferStringContainsOverIndexOfTitle,
+                        createChangedDocument: c => ReplaceBinaryOperationWithContains(doc, instanceOperation.Syntax, root, invocationOperation.Arguments, binaryOperation, c),
+                        equivalenceKey: MicrosoftNetCoreAnalyzersResources.PreferStringContainsOverIndexOfMessage),
+                    context.Diagnostics);
+                return;
+
+                async Task<Document?> ReplaceBinaryOperationWithContains(Document document, SyntaxNode syntaxNode, SyntaxNode treeRoot, ImmutableArray<IArgumentOperation> indexOfMethodArguments, IBinaryOperation binaryOperation, CancellationToken cancellationToken)
+                {
+                    DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                    SyntaxEditor syntaxEditor = new SyntaxEditor(treeRoot, document.Project.Solution.Workspace);
```

Comment:
`DocumentEditor` derives from `SyntaxEditor`, you don't need both. For most cases, using `SyntaxEditor` should suffice.

```suggestion
                    var editor = new SyntaxEditor(treeRoot, document.Project.Solution.Workspace);
```